### PR TITLE
policy: DNS Poller initializes error map on poll

### PR DESCRIPTION
**Pull request by raybejjani**
The DNS poller didn't initialize the dnsName -> error map used during a
DNS poll. This would cause it to assign values to a nil map on error,
crashing the daemon.

fixes 92243feb5df4956e87874b94ba77278d67ba3206

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4903)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by raybejjani** [comment]
I'll merge this once tests pass since I doubt any janitor particularly cares about reviewing this (and I should probably update CODEOWNERS for these files anyway)

**Pull request merged**
